rdsn/src/replica/replica.cpp

```diff
@@ -581,5 +581,10 @@ error_code replica::store_app_info(app_info &info, const std::string &path)
     return err;
 }
 
+int32_t replica::mutation_2pc_min_replica_count()
+{
+    return std::min(_options->mutation_2pc_min_replica_count, _app_info.max_replica_count - 1);
```

Comment:
Should this be compared as follows ?
```c++
return std::min({_options->mutation_2pc_min_replica_count, _app_info.max_replica_count - 1, 1});
```

Since for single replica `max_replica_count` is 1, then `std::min(_options->mutation_2pc_min_replica_count, _app_info.max_replica_count - 1)` will be 0.

Reply:
I retain _options->mutation_2pc_min_replica_count means that in the case: mutation_2pc_min_replica_count = 2, max_replica_count = 5, but the mim_2pc should equal with 2 but not 4, if you add `1` and then the mutation_2pc_min_replica_count is invalid.

But your case is right, so the function should:
```
assert(app_info.max_replica_count > 0);
return app_info.max_replica_count <=2 ? 1 : std::min(_options->mutation_2pc_min_replica_count, _app_info.max_replica_count - 1);
```

Do you think it is ok？

Reply:
Yeah, this made sense. However, another case is that the replica count is increased from 1 to 3: since `_options->mutation_2pc_min_replica_count` is originally set to 1, `std::min(_options->mutation_2pc_min_replica_count, _app_info.max_replica_count - 1)` will still return 1. Does this means we have to modify `_options->mutation_2pc_min_replica_count` ?

Reply:
Yeah, retain `_options->mutation_2pc_min_replica_count` is for user custom requirements, if user mark it `1` that means user hope just 1 replica is ok

Reply:
I suggest set `mutation_2pc_min_replica_count = 2` by default

Reply:
OK, I understand. I think in most cases setting `mutation_2pc_min_replica_count` to 2 is ok.

Is there a scenario that user sets `mutation_2pc_min_replica_count` to 3 while `max_replica_count` is 3 ? In this case the real `mutation_2pc_min_replica_count` will return 2. Should we handle this scenario ?

Reply:
Oh，great! Your suggestion is important.

So I modify like:
```c++
assert(app_info.max_replica_count > 0);
if (mutation_2pc_min_replica_count > 0) {//  >0 means use the user config
    return mutation_2pc_min_replica_count;
} else {// otherwise use the table max_replica_count
    return app_info.max_replica_count <=2 ? 1 :  _app_info.max_replica_count - 1
}

Reply:
I think both methods can work well with most cases. Maybe it's better to reserve `mutation_2pc_min_replica_count` for special cases, so we can adopt first method ? And while `mutation_2pc_min_replica_count` is not enabled, use N/2+1 instead (no special reason, since PacificA doesn't need quorum write) ?
```
assert(_app_info.max_replica_count > 0);
if (mutation_2pc_min_replica_count > 0) {//  >0 means use the user config
    return mutation_2pc_min_replica_count;
} else {// otherwise use the table max_replica_count
    return app_info.max_replica_count <=2 ? 1 :  _app_info.max_replica_count / 2 + 1;
}
```

Reply:
OK, I agree!